### PR TITLE
tech: add sendgrid api key

### PR DESCRIPTION
**Pull request by sabahirfan**
### Change description ###
This PR adds terraform configurations to pick up sendgrid api key.

**Does this PR introduce a breaking change?** (check one with "x")

- [ ] Yes
- [x] No

### CODE REVIEW

File: infrastructure/variables.tf [hunk 1]
```diff
@@ -1,5 +1,9 @@
 variable "product" {}
 
+variable "raw_product" {
```
Comment (timja) [inline infrastructure/variables.tf]: why do you need this?

File: infrastructure/main.tf [hunk 2]
```diff
@@ -22,9 +22,39 @@ module "key-vault" {
   create_managed_identity = true
 }
 
+locals {
+  vaultName = "${var.raw_product}-${var.env}"
```
Comment (timja) [inline infrastructure/main.tf]:
```suggestion
  vaultName = "${var.product}-${var.env}"
```

File: infrastructure/main.tf [hunk 3]
```diff
@@ -22,9 +22,39 @@ module "key-vault" {
...
+}
+
+data "azurerm_key_vault" "damages_key_vault" {
+  name = "${local.vaultName}"
```
Comment (timja) [inline infrastructure/main.tf]: you don't need string interpolation if you aren't interpolating
```suggestion
  name = local.vaultName
```

**Review by dharmendrak — DISMISSED** [review]

**Review by pd-dudkiewicz — DISMISSED** [review]

**Review by ckalista — APPROVED** [review]

**Review by ckalista — DISMISSED** [review]

File: infrastructure/variables.tf [hunk 4]
```diff
@@ -1,4 +1,6 @@
-variable "product" {}
+variable "product" {
+  default = "unspec"
+}
```
Comment (timja) [inline infrastructure/variables.tf]: don't set a default, it's set by the jenkins-library and these files often get copied around and out of sync
```suggestion
variable "product" {}
```

File: infrastructure/main.tf [hunk 5]
```diff
@@ -22,9 +22,39 @@ module "key-vault" {
   create_managed_identity = true
 }
 
+locals {
+  vaultName = "${var.product}-${var.env}"
+}
+
+data "azurerm_key_vault" "damages_key_vault" {
+  name = local.vaultName
+  resource_group_name = "${local.vaultName}"
```
Comment (timja) [inline infrastructure/main.tf]: quotes not required here
```suggestion
  resource_group_name = local.vaultName
```

**Review by timja — DISMISSED** [review]

**Review by ckalista — DISMISSED** [review]

**Review by dharmendrak — DISMISSED** [review]

**Review by dharmendrak — APPROVED** [review]

**Review by ckalista — APPROVED** [review]

**Pull request merged**